### DISCUSSION TITLE
Reducing HTTP overhead of the get-user-stateless endpoint

### DISCUSSION THREAD

**Discussion by farsadf** [deso-protocol/dips#31] [Medium CIPs]
# One-Line Description
Adding throttling to reduce the HTTP overhead of the `get-user-stateless` endpoint.

# Explanation and Motivation
As seen in the attachment below, the frontend sends a number of requests to the aforementioned endpoint; thus, creating an HTTP overhead (~2s per request is spent waiting for the first byte) for both the backend and the end-user. 
By implementing a simple throttling mechanism, we'll be merging such requests in one; thus, reducing the overhead costs of the `get-user-stateless` endpoint.

![Screen Shot 2021-07-10 at 9 25 37 PM](https://user-images.githubusercontent.com/5587389/125174373-61c46580-e1c5-11eb-9515-fec72c3c31ac.png)

**Comment by ItsAditya-xyz** [comment]
I think all those requests have their own specific use case, isn't it ?
I just checked the payload data passed to those endpoints and found that all those have different list public keys passed in `PublicKeysBase58Check`

**Reply by farsadf** [reply]
Yes indeed! 
They have different `PublicKeysBase58Check` values; however, the endpoint supports multiple public keys per request. With that said, we can create a simple stack structure for that specific endpoint and pushing into it whenever there are subsequent requests. Finally, execute one API call with all of the stack values and resolving the promises altogether.
